src/components/useAddToFavorites.js

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+import { useParams } from 'react-router-dom';
+
+export const useAddToFavorites = () => {
+  const addFavorites = "agregar a favoritos";
+  const removeFavorites = "eliminar de favoritos";
+  const add = "add";
+  const remove = "remove";
+  let { id } = useParams();
+  const getArray = JSON.parse(localStorage.getItem('favorites'));
```

Comment:
localStorage is not a recommended state management solution for react. I suggest you use context instead.
https://medium.com/@harshdigwani/redux-vs-context-api-vs-local-storage-bced5118c531